plugins/woocommerce-blocks/tests/e2e/tests/product-filters/product-filters.block_theme.spec.ts

```diff
@@ -158,5 +162,148 @@ test.describe( `${ blockData.name }`, () => {
+
+		// Layout settings
+		await expect(
+			editor.page.getByText( 'LayoutJustificationOrientation' )
+		).toBeVisible();
+	} );
+
+	test( 'Layout > default to vertical stretch', async ( {
+		editor,
+		pageObject,
+	} ) => {
+		await pageObject.addProductFiltersBlock( { cleanContent: true } );
+
+		const block = editor.canvas.getByLabel(
+			'Block: Product Filters (Experimental)'
+		);
+		await expect( block ).toBeVisible();
+
+		await editor.openDocumentSettingsSidebar();
+
+		const layoutSettings = editor.page.getByText(
+			'LayoutJustificationOrientation'
+		);
+		await expect(
+			layoutSettings.getByLabel( 'Justify items left' )
+		).not.toHaveAttribute( 'data-active-item' );
+		await expect(
+			layoutSettings.getByLabel( 'Stretch items' )
+		).toHaveAttribute( 'data-active-item' );
+		await expect(
+			layoutSettings.getByLabel( 'Horizontal' )
+		).not.toHaveAttribute( 'data-active-item' );
+		await expect( layoutSettings.getByLabel( 'Vertical' ) ).toHaveAttribute(
+			'data-active-item'
+		);
+	} );
+
+	test( 'Layout > Justification: changing option should update the preview', async ( {
+		editor,
+		pageObject,
+	} ) => {
+		await pageObject.addProductFiltersBlock( { cleanContent: true } );
+
+		const block = editor.canvas.getByLabel(
+			'Block: Product Filters (Experimental)'
+		);
+		await expect( block ).toBeVisible();
+
+		await editor.openDocumentSettingsSidebar();
+
+		const layoutSettings = editor.page.getByText(
+			'LayoutJustificationOrientation'
+		);
+		await layoutSettings.getByLabel( 'Justify items left' ).click();
+		await expect(
+			layoutSettings.getByLabel( 'Justify items left' )
+		).toHaveAttribute( 'data-active-item' );
+		await expect(
+			block.locator( blockData.selectors.editor.layoutWrapper )
+		).toHaveCSS( 'align-items', 'flex-start' );
+
+		await layoutSettings.getByLabel( 'Justify items center' ).click();
+		await expect(
+			layoutSettings.getByLabel( 'Justify items center' )
+		).toHaveAttribute( 'data-active-item' );
+		await expect(
+			block.locator( blockData.selectors.editor.layoutWrapper )
+		).toHaveCSS( 'align-items', 'center' );
+	} );
+
+	test( 'Layout > Orientation: changing option should update the preview', async ( {
+		editor,
+		pageObject,
+	} ) => {
+		await pageObject.addProductFiltersBlock( { cleanContent: true } );
+
+		const block = editor.canvas.getByLabel(
+			'Block: Product Filters (Experimental)'
+		);
+		await expect( block ).toBeVisible();
+
+		await editor.openDocumentSettingsSidebar();
+
+		const layoutSettings = editor.page.getByText(
+			'LayoutJustificationOrientation'
+		);
+		await layoutSettings.getByLabel( 'Horizontal' ).click();
+		await expect(
+			layoutSettings.getByLabel( 'Stretch items' )
+		).toBeHidden();
+		await expect(
+			layoutSettings.getByLabel( 'Space between items' )
+		).toBeVisible();
+		await expect(
+			block.locator( blockData.selectors.editor.layoutWrapper )
+		).toHaveCSS( 'flex-direction', 'row' );
```

Comment:
I'm not happy with CSS locator and assertion, but it seems to be the only way to test these settings AFAIK. I'm open to any suggestions.

Reply:
One option would be to use css locators based on layout position, like in these [examples](https://playwright.dev/docs/other-locators#css-matching-elements-based-on-layout):

```
// Fill an input to the right of "Username".
await page.locator('input:right-of(:text("Username"))').fill('value');

// Click a button near the promo card.
await page.locator('button:near(.promo-card)').click();

// Click the radio input in the list closest to the "Label 3".
await page.locator('[type=radio]:left-of(:text("Label 3"))').first().click();
```

Reply:
Ah, I checked your [suggestion](https://github.com/woocommerce/woocommerce/pull/48429#pullrequestreview-2126170020) but I haven't gotten it until now. We can check if the text Status which is right of Price is visible or not to verify the layout direction. Thank you, I will try it in this PR.

About the block spacing tests, do you have any alternative?